test/strings.js

```diff
@@ -59,6 +59,22 @@ describe('strings', () => {
       expect(inspect('foobarbaz', { truncate: 3 })).to.equal("'…'")
     })
 
+    it('truncates strings involving surrogate pairs longer than truncate (7)', () => {
+      expect(inspect('🐱🐱🐱', { truncate: 7 })).to.equal("'🐱🐱…'") // not '🐱🐱\ud83d…'
+    })
+
+    it('truncates strings involving surrogate pairs longer than truncate (6)', () => {
+      expect(inspect('🐱🐱🐱', { truncate: 6 })).to.equal("'🐱…'") // not '🐱\ud83d…'
+    })
+
+    it('truncates strings involving surrogate pairs longer than truncate (5)', () => {
+      expect(inspect('🐱🐱🐱', { truncate: 5 })).to.equal("'🐱…'")
+    })
+
+    it('truncates strings involving graphemes than truncate (5)', () => {
```

Comment:
Supporting graphemes, is way more complex than surrogate pairs. As such splitting a grapheme (aka a visual unit from a user point of view) is acceptable for Unicode as the string remains valid. On the opposite splitting a surrogate pair is responsible to create strings that are considered invalid from an Unicode point of view as they contain illegal characters.